File: internal/utils/scale.go
package utils

func Scale(n float64, lo float64, hi float64, tlo float64, thi float64) float64 {
	if n < lo {
		n = lo
	}
	if n > hi {
		n = hi
	}
	return (n-lo)/(hi-lo)*(thi-tlo) + tlo
}

File: web/src/@utils/scale.ts
export function scale(n: number, lo: number, hi: number, tlo: number, thi: number): number {
  if (n < lo) {
    n = lo;
  }
  if (n > hi) {
    n = hi;
  }
  return (n - lo) / (hi - lo) * (thi - tlo) + tlo;
}
